infra/vsts_agent_ubuntu_20_04.tf

```diff
@@ -46,18 +46,10 @@ resource "google_compute_instance_template" "vsts-agent-ubuntu_20_04" {
     create_before_destroy = true
   }
 
-  metadata {
-    startup-script = "${data.template_file.vsts-agent-ubuntu_20_04-startup.rendered}"
+  metadata = {
+    startup-script = data.template_file.vsts-agent-ubuntu_20_04-startup.rendered
 
-    shutdown-script = <<EOS
-#!/usr/bin/env bash
-set -euo pipefail
-cd /home/vsts/agent
-su vsts <<SHUTDOWN_AGENT
-export VSTS_AGENT_INPUT_TOKEN='${secret_resource.vsts-token.value}'
-./config.sh remove --unattended --auth PAT
-SHUTDOWN_AGENT
-    EOS
+    shutdown-script = "#!/usr/bin/env bash\nset -euo pipefail\ncd /home/vsts/agent\nsu vsts <<SHUTDOWN_AGENT\nexport VSTS_AGENT_INPUT_TOKEN='${secret_resource.vsts-token.value}'\n./config.sh remove --unattended --auth PAT\nSHUTDOWN_AGENT\n    "
```

Comment:
Because the existing heredoc was, technically, malformed, there does not seem to exist a new heredoc syntax that produces the same output as before. Specifically, it does not seem to be possible to end a heredoc on spaces (vs. a newline) anymore.

This is probably not the end state we want, but this was not the point of this PR and adding a newline here requires taking down the entire CI cluster, which seemed like a bad idea while people are working.